clients/src/test/java/org/apache/kafka/common/internals/PluginTest.java

```diff
@@ -0,0 +1,124 @@
+        Plugin<SomeMonitorablePlugin> pluginMonitorable = Plugin.wrapInstance(someMonitorablePlugin, METRICS, CONFIG);
+        checkPlugin(pluginMonitorable, someMonitorablePlugin, true);
+
+        someMonitorablePlugin = new SomeMonitorablePlugin();
+        assertFalse(someMonitorablePlugin.closed);
+        pluginMonitorable = Plugin.wrapInstance(someMonitorablePlugin, null, CONFIG);
+        checkPlugin(pluginMonitorable, someMonitorablePlugin, false);
+
+        SomePlugin somePlugin = new SomePlugin();
+        assertFalse(somePlugin.closed);
+        Plugin<SomePlugin> plugin = Plugin.wrapInstance(somePlugin, null, CONFIG);
+        assertSame(somePlugin, plugin.get());
+        assertNull(somePlugin.pluginMetrics);
+        plugin.close();
+        assertTrue(somePlugin.closed);
+    }
+
+    @Test
+    void testWrapInstances() throws Exception {
+        List<SomeMonitorablePlugin> someMonitorablePlugins = Arrays.asList(new SomeMonitorablePlugin(), new SomeMonitorablePlugin());
+        List<Plugin<SomeMonitorablePlugin>> pluginsMonitorable = Plugin.wrapInstances(someMonitorablePlugins, METRICS, CONFIG);
+        assertEquals(someMonitorablePlugins.size(), pluginsMonitorable.size());
+        for (int i = 0; i < pluginsMonitorable.size(); i++) {
+            Plugin<SomeMonitorablePlugin> plugin = pluginsMonitorable.get(i);
+            SomeMonitorablePlugin somePlugin = someMonitorablePlugins.get(i);
+            checkPlugin(plugin, somePlugin, true);
+        }
+
+        someMonitorablePlugins = Arrays.asList(new SomeMonitorablePlugin(), new SomeMonitorablePlugin());
+        pluginsMonitorable = Plugin.wrapInstances(someMonitorablePlugins, null, CONFIG);
+        assertEquals(someMonitorablePlugins.size(), pluginsMonitorable.size());
+        for (int i = 0; i < pluginsMonitorable.size(); i++) {
+            Plugin<SomeMonitorablePlugin> plugin = pluginsMonitorable.get(i);
+            SomeMonitorablePlugin somePlugin = someMonitorablePlugins.get(i);
+            checkPlugin(plugin, somePlugin, false);
+        }
+
+        List<SomePlugin> somePlugins = Arrays.asList(new SomePlugin(), new SomePlugin());
+        List<Plugin<SomePlugin>> plugins = Plugin.wrapInstances(somePlugins, METRICS, CONFIG);
+        assertEquals(somePlugins.size(), plugins.size());
+        for (int i = 0; i < plugins.size(); i++) {
+            Plugin<SomePlugin> plugin = plugins.get(i);
+            SomePlugin somePlugin = somePlugins.get(i);
+            assertSame(somePlugin, plugin.get());
+            assertNull(somePlugin.pluginMetrics);
+            plugin.close();
+            assertTrue(somePlugin.closed);
+        }
+    }
+
+    private void checkPlugin(Plugin<SomeMonitorablePlugin> plugin, SomeMonitorablePlugin instance, boolean metricsSet) throws Exception {
+        assertSame(instance, plugin.get());
+        if (metricsSet) {
+            assertNotNull(instance.pluginMetrics);
```

Comment:
I think we should also make sure the content of the `pluginMetrics` is expected. Ex: we can add some sensors, tags and verify they are added in the pluginMetrics.

Reply:
We do that in `PluginMetricsImplTest` and in the test classes for each supported plugin. So I thought focusing this test class on the wrapping and closing of plugin instances made more sense.

Reply:
OK. Thanks.